datafusion/core/tests/dataframe/mod.rs

```diff
@@ -212,7 +212,6 @@ async fn test_count_wildcard_on_aggregate() -> Result<()> {
     let sql_results = ctx
         .sql("select count(*) from t1")
         .await?
-        .select(vec![col("count(*)")])?
```

Comment:
There seems to be a reason for doing it; I once wanted to remove it as well 😂. 
See https://github.com/apache/datafusion/pull/10459#discussion_r1598937560

Reply:
Cool, but I still don't get it.

> Perhaps it meant to show how to select the count(*) again -- if so maybe we could update the tests to do something like this (rather than removing it entirely):

1. Why do we need to select count again, is it for test that showing select twice has the correct result?
2. Why select with `count(*)` but not `count(wildcard())`?

For `count(*)`, it directly goes to PlanBuilder and miss the chance for ExprPlanner rewrite. IMO we should use `count(wildcard())` instead but consider `select(vec![col("count(*)")])` invalid 🤔

Reply:
I think the idea was to show that it is possible to select a column named `"count(*)"` after computing the actual `count(*)` value in a previous plan. 

```text
Projection(expr=[col("count(*)")]
  Aggregate(agg=[count(*))
   Scan(..)
```


If we can remove the code and the test still passes though clearly something is not working as intended. 

Maybe it should be instead `col("count(*)") + lit(1)` so the expression doesn't get removed by the optimizer 🤔

Reply:
😕